### PR TITLE
Token definitions moved from wallet state to root state

### LINKED ISSUE #11506: Move tokenDefinitions to root redux store from wallet
https://github.com/trezor/trezor-suite/pull/11484#discussion_r1516533115

### PR DESCRIPTION
**Pull request by tomasklim**
## Related Issue

closes #11506

## Screenshots:
Screenshot 2024-05-03 at 11 24 57
Screenshot 2024-05-03 at 11 25 26

### CODE REVIEW

**Comment by tomasklim** [comment]
@trezor/suite-mobile would be nice if you check that it still works on mobile. The main reason is this https://github.com/trezor/trezor-suite/pull/12250/commits/46ed9800ad381f6a26e0f0e40df677eec4cddb49

File: suite-common/wallet-core/src/fiat-rates/fiatRatesThunks.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { getNetworkFeatures } from '@suite-common/wallet-config';
 import { FIAT_RATES_MODULE_PREFIX, REFETCH_INTERVAL } from './fiatRatesConstants';
 import { selectTickersToBeUpdated, selectTransactionsWithMissingRates } from './fiatRatesSelectors';
 import { transactionsActions } from '../transactions/transactionsActions';
-import { selectIsSpecificCoinDefinitionKnown } from '../token-definitions/tokenDefinitionsSelectors';
+import { selectIsSpecificCoinDefinitionKnown } from '@suite-common/token-definitions/src/tokenDefinitionsSelectors';
```
Comment (PeKne) [inline suite-common/wallet-core/src/fiat-rates/fiatRatesThunks.ts]:
```suggestion
import { selectIsSpecificCoinDefinitionKnown } from '@suite-common/token-definitions';
```

File: suite-common/wallet-core/src/transactions/transactionsReducer.ts [hunk 2]
```diff
@@ -16,8 +16,8 @@ import {
     selectBlockchainHeightBySymbol,
     BlockchainRootState,
 } from '../blockchain/blockchainReducer';
-import { selectNetworkTokenDefinitions } from '../token-definitions/tokenDefinitionsSelectors';
-import { TokenDefinitionsRootState } from '../token-definitions/tokenDefinitionsTypes';
+import { selectNetworkTokenDefinitions } from '@suite-common/token-definitions/src/tokenDefinitionsSelectors';
```
Comment (PeKne) [inline suite-common/wallet-core/src/transactions/transactionsReducer.ts]:
```suggestion
import { selectNetworkTokenDefinitions } from '@suite-common/token-definitions';
```

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ import { selectTransactions } from './transactionsReducer';
 import { TRANSACTIONS_MODULE_PREFIX, transactionsActions } from './transactionsActions';
 import { selectAccountByKey, selectAccounts } from '../accounts/accountsReducer';
 import { selectBlockchainHeightBySymbol } from '../blockchain/blockchainReducer';
-import { selectNetworkTokenDefinitions } from '../token-definitions/tokenDefinitionsSelectors';
+import { selectNetworkTokenDefinitions } from '@suite-common/token-definitions/src/tokenDefinitionsSelectors';
```
Comment (PeKne) [inline suite-common/wallet-core/src/transactions/transactionsThunks.ts]:
```suggestion
import { selectNetworkTokenDefinitions } from '@suite-common/token-definitions';
```

File: suite-native/discovery/src/discoverySelectors.ts [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
-} from '@suite-common/wallet-core';
+    selectValidTokensByDeviceStateAndNetworkSymbol,
+} from '@suite-common/token-definitions';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { DeviceRootState } from '@suite-common/wallet-core';
 import {
     AccountsRootState,
+    selectAccountsByDeviceStateAndNetworkSymbol,
     selectDeviceAccounts,
 } from '@suite-common/wallet-core/src/accounts/accountsReducer';
```
Comment (PeKne) [inline suite-native/discovery/src/discoverySelectors.ts]: this import is wrong. I know it is not your mistake, but could you fix it aswell please?
```suggestion
} from '@suite-common/wallet-core';
```

**Review by PeKne — CHANGES_REQUESTED** [review]
Looks good. Just a few malformed imports.

File: suite-native/discovery/src/discoverySelectors.ts (hunk 4 above)
Comment (tomasklim) [reply suite-native/discovery/src/discoverySelectors.ts]: @PeKne All imports fixed, I unfortunately squashed it

**Review by PeKne — APPROVED** [review]

**Pull request merged**